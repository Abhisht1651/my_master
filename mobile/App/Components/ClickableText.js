// @flow
import React from 'react'
import { Text, TouchableOpacity, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'

type Props = {
  onPress?: PropTypes.func,
  textStyle?: StyleSheet,
  children?: string,
  style?: StyleSheet,
  text?: string
}
const ClickableText = (props: Props) => {
  const { style, onPress, textStyle, children, text } = props
  return (
    <TouchableOpacity style={style} onPress={onPress}>
      <Text style={textStyle}> {children || text} </Text>
    </TouchableOpacity>
  )
}

ClickableText.defaultProps = {
  onPress: () => {},
  textStyle: StyleSheet.create({}),
  children: '',
  style: StyleSheet.create({}),
  text: ''
}

export default ClickableText
