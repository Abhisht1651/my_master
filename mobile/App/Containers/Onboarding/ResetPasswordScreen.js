/* eslint-disable no-unused-vars */
// @flow
import { NavigationContainerProps } from '@react-navigation/native'

import React, { useState } from 'react'
import { Text, Image, View, TouchableOpacity, Platform } from 'react-native'
import { Images } from '../../Themes'

// Styles
import styles from '../Styles/Onboarding/ResetPasswordScreenStyles'
import ClickableText from '../../Components/ClickableText'
import PasswordText from '../../Components/TextInput/TextInputForPassword'
import { MediumButton } from '../../Components/Buttons/MediumButton'

type Props = NavigationContainerProps & {}
const ResetPasswordScreen = (props: Props) => {
  const [passwordtext, setpasswordtext] = useState('')
  const [confirmPassword, setconfirmPassword] = useState('')
  const [password, setpassword] = useState('')
  const [type, settype] = useState(-1)
  const [strength, setstrength] = useState('')
  const [isfocussed, setisfocussed] = useState(true)

  const performValidation = (val: string, toValdidate: boolean, validationType: string) => {
    let changeText = ''
    let stren = ''
    let typee = -1
    const pattern = [
      '^.{8,}$', // min 8 chars
      '(?=.*\\d)', // number required
      '(?=.*[A-Z])', // uppercase letter
      '(?=.*[!@#$%^&*])' // special Character
    ]
    const conditions = pattern.map((rule) => new RegExp(rule, 'g'))
    if (val === '') {
      setpasswordtext('')
      setconfirmPassword('')
      setpassword('')
      settype(-1)
      setstrength('')
      return
    }

    if (toValdidate) {
      switch (validationType) {
        case 'password': {
          const checkValid = conditions.map((condition) => condition.test(val))
          // console.log(checkValid);

          if (!checkValid[0]) {
            changeText = 'Password must be minimum 8 characters'
          } else if (!checkValid[1] || !checkValid[2]) {
            stren = 'Weak'
            changeText = 'try using numbers or capital letters'
            typee = 0
          } else if (!checkValid[3]) {
            stren = 'Good'
            typee = 1
            changeText = 'not bad you can go ahead'
          } else {
            stren = 'Strong'
            changeText = ''
            typee = 2
          }
          setpasswordtext(changeText)
          setpassword(val)
          settype(typee)
          setstrength(stren)
          // console.log(state);

          break
        }
        default:
          break
      }
    }
    // console.log(state);
  }

  const onFocus = () => {
    setisfocussed(true)
  }

  const onBlur = () => {
    setisfocussed(false)
  }

  const checkpassword = () => {
    return password === confirmPassword && password !== '' && confirmPassword !== ''
  }
  const getStyle = (typee: number = -1) => {
    if (typee === 0) return styles.weak
    if (typee === 1) return styles.good
    return styles.strong
  }

  const getColor = (typee: number = -1) => {
    if (typee > -1) return { color: '#8F8FAF' }
    return undefined
  }

  return (
    <View style={styles.mainContainer}>
      {Platform.OS === 'ios' && <View style={styles.statusBar} />}
      <View style={styles.mainContainer}>
        <TouchableOpacity
          onPress={() => props.navigation.goBack()}
          style={styles.backArrowContainer}>
          <Image source={Images.leftArrow} style={styles.backArrow} />
        </TouchableOpacity>
        <View style={styles.section}>
          <Image source={Images.newzeraPurpleLogo} style={styles.logo} />
          <Text style={styles.resetPasswordText}>Reset Password</Text>
          <View style={[styles.topsectionMargin, styles.TextInputWidth]}>
            <PasswordText
              placeholder="New Password"
              captureText={(val) => performValidation(val, true, 'password')}
            />
            {checkpassword() === true && (
              <Image source={Images.checkMark} style={styles.checkmark} />
            )}
            <View style={styles.warningContainer}>
              <View style={styles.dirRow}>
                <Text style={getStyle(type)}>{strength} </Text>
                <Text style={[styles.warningText, getColor(type)]}>{passwordtext}</Text>
                {type === 2 && <Image source={Images.checkMark} style={styles.checkmark} />}
              </View>
            </View>
          </View>
          <View style={[styles.textMarginWarning, styles.TextInputWidth]}>
            <PasswordText
              placeholder="Confirm Password"
              captureText={(val) => {
                setconfirmPassword(val)
              }}
            />
            {checkpassword() === true && (
              <Image source={Images.checkMark} style={styles.checkmark} />
            )}
          </View>
        </View>
        <View style={styles.buttonWrap}>
          <MediumButton title="Continue" onPress={() => props.navigation.navigate('LoginScreen')} />

          <View style={styles.sectionRow}>
            <Text style={styles.signUpTextMain}>Not an user? </Text>
            <ClickableText
              textStyle={styles.signUpTextSignUp}
              onPress={() => props.navigation.navigate('SignUp')}
              text="Sign up"
            />
          </View>
        </View>
      </View>
    </View>
  )
}
export default ResetPasswordScreen
