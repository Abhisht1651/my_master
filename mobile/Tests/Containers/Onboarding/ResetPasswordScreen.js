import { Image, Text } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import ResetPasswordScreen from '../../../App/Containers/Onboarding/ResetPasswordScreen'
import PasswordText from '../../../App/Components/TextInput/TextInputForPassword'
import { MediumButton } from '../../../App/Components/Buttons/MediumButton'
import ClickableText from '../../../App/Components/ClickableText'

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
})

describe('ResetPasswordScreen', () => {
  describe('rendering', () => {
    let wrapper
    let props
    beforeEach(() => {
      props = createTestProps({})
      wrapper = shallow(<ResetPasswordScreen {...props} />) // no compile-time error
    })
    it('should match to snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render a <Image />', () => {
      expect(wrapper.find(Image)).toHaveLength(2)
    })

    it('should render 2 <PasswordText />', () => {
      expect(wrapper.find(PasswordText)).toHaveLength(2)
    })
    it('<PasswordText /> is password', () => {
      const inputText = wrapper.find(PasswordText).at(0)
      expect(inputText.prop('placeholder')).toBe('New Password')
    })
    it('<PasswordText /> at 1 is confirm password', () => {
      const inputText = wrapper.find(PasswordText).at(1)
      expect(inputText.prop('placeholder')).toBe('Confirm Password')
    })
    it('should render 4 <Text /> tag', () => {
      expect(wrapper.find(Text)).toHaveLength(4)
    })

    it('should render 1 <Rounded Button />', () => {
      expect(wrapper.find(MediumButton)).toHaveLength(1)
    })

    it('should render <ClickableText /> tag', () => {
      expect(wrapper.find(ClickableText)).toHaveLength(1)
    })
    it('should navigate to SignUp Screen when clicked on SignUp Text', () => {
      const button1 = wrapper.find(ClickableText).at(0)
      button1.simulate('press')
      expect(props.navigation.navigate).toHaveBeenCalledWith('SignUp')
    })
  })
})
