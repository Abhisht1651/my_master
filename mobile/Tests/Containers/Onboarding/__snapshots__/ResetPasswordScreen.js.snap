// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordScreen rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.188802083333336,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "left": 54,
          "position": "absolute",
          "top": 2.60546875,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 31.265625,
            "resizeMode": "contain",
            "width": 36,
          }
        }
      />
    </ForwardRef>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 217.04179999999997,
            "marginTop": 133.4,
            "resizeMode": "contain",
            "width": 434.99999999999994,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Medium",
            "fontSize": 38,
            "marginTop": 38.686,
          }
        }
      >
        Reset Password
      </Text>
      <View
        style={
          Array [
            Object {
              "marginTop": 142.43229166666669,
            },
            Object {
              "width": 550,
            },
          ]
        }
      >
        <TextInputForPassword
          borderBottomColor="#7B46E4"
          borderBottomStyle={Object {}}
          captureText={[Function]}
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="New Password"
          placeholderTextColor="#8F8FAF"
        />
        <View
          style={
            Object {
              "height": 24.317708333333336,
              "justifyContent": "center",
              "marginTop": 8.684895833333334,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#08C981",
                  "fontFamily": "BananaGrotesk-Medium",
                  "fontSize": 28,
                  "textAlign": "center",
                }
              }
            >
               
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FF1742",
                    "fontFamily": "BananaGrotesk-Regular",
                    "fontSize": 25,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 26.0546875,
            },
            Object {
              "width": 550,
            },
          ]
        }
      >
        <TextInputForPassword
          borderBottomColor="#7B46E4"
          borderBottomStyle={Object {}}
          captureText={[Function]}
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="Confirm Password"
          placeholderTextColor="#8F8FAF"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 86.84895833333334,
        }
      }
    >
      <MediumButton
        disabled={false}
        onPress={[Function]}
        outline={false}
        style={Object {}}
        title="Continue"
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": "23%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 31,
            }
          }
        >
          Not an user? 
        </Text>
        <ClickableText
          onPress={[Function]}
          style={Object {}}
          text="Sign up"
          textStyle={
            Object {
              "color": "#7B46E4",
              "fontFamily": "BananaGrotesk-Bold",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
